File: red-scare/none.py
import math
import numpy as np
#import networkx as nx

class NoneGraph:
    def __init__(self, adjMatrix, isRed, n):
        self.graph = np.zeros((n, n), dtype=int)
        self.n = n

        for i in range(int(self.n)):
            for j in range(int(self.n)):
                if isRed[i] or isRed[j]:
                    continue
#                if adjMatrix[i,j] == 1:
#                    self.G.add_edge(i,j)
                self.graph[i,j] = adjMatrix[i,j]



    def minDistance(self, dist, sptSet):
        minimum = math.inf

        for u in range(self.n):
            if dist[u] < minimum and sptSet[u] == False:

                minimum = dist[u]
                min_index = u

        return min_index

    def dijkstra(self, s, t):
        dist = [math.inf] * self.n
        dist[s] = 0

        sptSet = [False] * self.n

        for cout in range(self.n):
            try:
                x = self.minDistance(dist, sptSet)
            except UnboundLocalError: # The node is disconnected
                continue 

            sptSet[x] = True
            for y in range(self.n):
                if self.graph[x,y] > 0 and sptSet[y] == False and dist[y] > dist[x] + self.graph[x,y]:
                    
                    dist[y] = dist[x] + self.graph[x,y]

        return dist[t]


#    def shortestPath(self, s, t):
#        return len(nx.shortest_path(self.G, source=s, target=t))-1

File: red-scare/src/none.go
package main

import "math"

func minDistance(dist []float64, sptSet []bool, n int) int {
	var minIndex int
	minimum := math.Inf(1)

	for i := 0; i < n; i++ {
		if dist[i] < minimum && !sptSet[i] {
			minimum = dist[i]
			minIndex = i
		}
	}
	return minIndex
}

func (g *graph) dijkstra(reds []bool, s int, t int, n int) int {

	dist := make([]float64, n)
	for i := range dist {
		dist[i] = math.Inf(1)
	}
	dist[s] = 0

	sptSet := make([]bool, n)

	for i := 0; i < n; i++ {

		x := minDistance(dist, sptSet, n)

		sptSet[x] = true

		for j := 0; j < n; j++ {
			if (*g)[i][j] > 0 && !sptSet[j] && dist[j] > dist[i]+float64((*g)[i][j]) {
				dist[j] = dist[i] + float64((*g)[i][j])
			}
		}

	}

	if dist[t] != math.Inf(1) {
		return int(dist[t])
	} else {
		return -1
	}

}
